### PR TITLE
attempt at listing some system rules with examples

### PR DESCRIPTION
**Pull request by tariqkurd-repo**
Something which came up in RVI discussions is how to build CHERI systems, so this section adds some system guidelines.

### CODE REVIEW

File: src/system.adoc [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+.. These busses will write the tag to memory as an extension of the data write.
+. Non-tag aware busses, i.e. normal non-CHERI aware busses.
+.. Reads of tagged memory will not read the tag.
+.. Writes to tagged memory will clear the tag of any CLEN-aligned CLEN-wide memory location where any byte matches the memory write.
+
+The fundamental rule for any CHERI system is that the tag and data are always accessed atomically. For every CLEN-aligned CLEN-wide memory location in memory It must never be possible to:
+
+. Update any data bytes without also writing the tag
+.. This implies clearing the tag if a non-CHERI aware bus master overwrites a capability in memory
+. Set the tag without also writing the data.
```
Comment (arichardson) [inline src/system.adoc:21]:
```suggestion
. Read a tagged value with mismatched (stale or newer) data
. Set the tag without also writing the data.
```
This is implicit in the "accessed atomically" but might be worth calling out explicitly?

File: src/system.adoc [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+.. Clearing tags in memory does not require updating the data
```
Comment (arichardson) [inline src/system.adoc:22]: Maybe this works better as a note?
```suggestion

NOTE: Clearing tags in memory does not necessarily require updating the associated data.
```

File: src/system.adoc [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+There are two types of bus connections used in chips which contain CHERI CPUs:
+
+. Tag-aware busses, where the bus protocol is extended to carry the tag along with the data.  This is typically done using a user defined bit in the protocol.
+.. These busses will read tags from memory (if tags are present in the target memory) and return them to the requestor.
+.. These busses will write the tag to memory as an extension of the data write.
+. Non-tag aware busses, i.e. normal non-CHERI aware busses.
+.. Reads of tagged memory will not read the tag.
+.. Writes to tagged memory will clear the tag of any CLEN-aligned CLEN-wide memory location where any byte matches the memory write.
+
+The fundamental rule for any CHERI system is that the tag and data are always accessed atomically. For every CLEN-aligned CLEN-wide memory location in memory It must never be possible to:
```
Comment (andresag01) [inline src/system.adoc:17]:
```suggestion
The fundamental rule for any CHERI system is that the tag and data are always accessed atomically. For every naturally aligned CLEN-wide memory location, it must never be possible to:
```

**Review by arichardson — COMMENTED** [review]
Had some outstanding comments that I forgot to submit but nothing major.

File: src/system.adoc [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+CHERI processors need memory systems which support the capability validity tags in memory.
+
+There are, or will soon be, a wide range of CHERI systems in existence from tiny IoT devices up to server chips.
+
+There are two types of bus connections used in chips which contain CHERI CPUs:
+
+. Tag-aware busses, where the bus protocol is extended to carry the tag along with the data.  This is typically done using a user defined bit in the protocol.
+.. These busses will read tags from memory (if tags are present in the target memory) and return them to the requestor.
+.. These busses will write the tag to memory as an extension of the data write.
+. Non-tag aware busses, i.e. normal non-CHERI aware busses.
```
[the remark this answers is not in the document]
Comment (tariqkurd-repo) [reply src/system.adoc:13]: I've put these into https://github.com/riscv/riscv-cheri/pull/409

**Pull request merged**